src/main/kotlin/org/rust/ide/refactoring/inlineFunction/RsInlineFunctionProcessor.kt

```diff
@@ -209,7 +217,50 @@ class RsInlineFunctionProcessor(
         }
     }
 
-    private fun addFunctionBodyToCallerWithCorrectSpacing(leftSibling: PsiElement?, rightSibling: PsiElement?, blockParent: RsBlock, body: RsBlock, childContainingEnclosingStatement: PsiElement?) {
+    private fun replaceParameterUsages(functionDup: RsFunction, caller: PsiElement, funcScope: LocalSearchScope) {
+        val parameterBindingsList = functionDup.valueParameters
+            .filter { it !is RsSelfParameter }
+            .map { it.pat }
+            .filterIsInstance(RsPat::class.java)
+
+        val valueArgumentList = when (caller) {
+            is RsCallExpr -> caller.valueArgumentList
+            is RsDotExpr -> caller.methodCall?.valueArgumentList ?: return
+            else -> return
+        }
+        for ((index, parameter) in parameterBindingsList.withIndex()) {
+            var argument = valueArgumentList.exprList[index]
+            if (argument is RsUnaryExpr) {
```

Comment:
Why is this here? The following case produced wrong output:
```rust
fn foo(x: bool) {
    println!("{}", x);
}
fn main() {
    foo/*caret*/(!true);
}
```

Reply:
It's supposed to remove `mut` and `&` (mut is handled seperately if needed, & is automatically dereferenced). I'll fix the if statement accordigly. It helps about 10 tests reach the state where `name == argument.text` and there's not *let* decleration creation.

Reply:
Fixed, now it it will only shorten the argument if all leafs are `&` or `mut`.

Reply:
It still breaks basics scenarios:
```rust
fn foo(x: &i32) {
    println!("{}", *x);
}
fn main() {
    let a = 5;
    foo/*caret*/(&a);
}
```

You can't just arbitrarily remove unary expressions, therefore I suggest to remove this code entirely :)